src/client/components/ModalNew.tsx

```diff
@@ -95,6 +95,18 @@ export const Modal = ({
 
   const hasOnCloseFunction = Boolean(onClose)
 
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === 'Escape' && !!onClose) {
```

Comment:
Using `hasOnCloseFunction` above doesn't satisfy TS 🤷

Reply:
What if it were ```const hasOnCloseFunction = typeof onClose === 'function'```?  

It might be that Boolean does not narrow it enough

Reply:
Yes, nice, that worked ⭐

Reply:
erhm.. Worked i VSCode but tsc failed it when trying to push :trollface: